src/integrationTest/java/com/mongodb/hibernate/type/temporal/InstantIntegrationTests.java

```diff
@@ -0,0 +1,229 @@
+                    Arguments.of(
+                            tz0,
+                            tz1,
+                            // Value to save.
+                            // We support milliseconds precision, so nanoseconds are rounded down to milliseconds.
+                            Instant.parse("2007-12-03T10:15:30.002900000Z"),
+                            // Expected value.
+                            Instant.parse("2007-12-03T10:15:30.002000000Z")),
+                    Arguments.of(
+                            tz0, tz1, Instant.parse("1500-12-03T10:15:30Z"), Instant.parse("1500-12-03T10:15:30Z")),
+                    Arguments.of(
+                            tz0,
+                            tz1,
+                            Instant.parse("-000001-12-03T10:15:30Z"),
+                            Instant.parse("-000001-12-03T10:15:30Z")));
+        });
+    }
+
+    @ParameterizedTest(name = "testRoundTripSessionTzsEqual: Write(sys={0}, sess={1}). Read(sys={0}, sess={1})")
+    @MethodSource("instantPersistAndReadParameters")
+    void testRoundTripSessionTzsEqual(ZoneId systemDefaultTimeZone, ZoneId jdbcTimeZone, Instant toSave, Instant toRead)
+            throws Exception {
+        var instantItem = new Item(1, toSave);
+        withSystemTimeZone(
+                systemDefaultTimeZone, () -> inTransaction(jdbcTimeZone, session -> session.persist(instantItem)));
+        var loadedInstantItem = withSystemTimeZone(
+                systemDefaultTimeZone,
+                () -> fromTransaction(jdbcTimeZone, session -> session.find(Item.class, instantItem.id)));
+
+        var expectedItem = new Item(1, toRead);
+        assertEq(expectedItem, loadedInstantItem);
+    }
+
+    @ParameterizedTest(
+            name = "testRoundTripWriteAndReadPathTzsNotEqual: Write(sys={0}, sess={0}). Read(sys={1}, sess={1})")
+    @MethodSource("instantPersistAndReadParameters")
+    void testRoundTripWriteAndReadPathTzsNotEqual(
+            ZoneId writeTimeZonePath, ZoneId readTimeZonePath, Instant toSave, Instant toRead) throws Exception {
+        var instantItem = new Item(1, toSave);
+        withSystemTimeZone(
+                writeTimeZonePath, () -> inTransaction(writeTimeZonePath, session -> session.persist(instantItem)));
+        var loadedInstantItem = withSystemTimeZone(
+                readTimeZonePath,
+                () -> fromTransaction(readTimeZonePath, session -> session.find(Item.class, instantItem.id)));
+
+        var expectedItem = new Item(1, toRead);
+        assertEq(expectedItem, loadedInstantItem);
+    }
+
+    @ParameterizedTest(
+            name = "testRoundTripSessionTzsNotEqual:" + "Write(sys=" + ZONE_OFFSET + ", sess={0}). Read(sys="
+                    + ZONE_OFFSET + ", sess={1})")
+    @MethodSource("instantPersistAndReadParameters")
+    void testRoundTripSessionTzsNotEqual(
+            ZoneId sessionWriteTimeZone, ZoneId sessionReadTimeZone, Instant toSave, Instant toRead) throws Exception {
```

Comment:
The names `sessionWriteTimeZone`/`sessionReadTimeZone` are used here, but the name `jdbcTimeZone` is used in `testRoundTripSessionTzsEqual`. The naming should be consistent.